### DISCUSSION TITLE
moving from semantic to fomantic - how to?

### DISCUSSION THREAD

**Discussion by birthdaycorp** [fomantic/Fomantic-UI#1889] [General] [answered]
Hi everyone,

I'm moving from the most recent version of semantic-ui to fomantic-ui, and am wondering if there is any additional set up beyond running `npm install fomantic-ui`? I ask because, to get semantic-ui to work in my Rails app, I've been relying on [this answer](https://stackoverflow.com/a/50086486/2009652). Note the additional files and steps. Now that I'm using fomantic, I get a `Less::Error Unrecognised input` when the assets in the page head start loading. Would this have anything to do with the extra set up involved? Do I still need **semantic.json**? I can see in the assets src folder that pretty much everything is the same, except for some updates and a "fomantic ui" at the top.

**Comment by lubber-de** [comment, marked as answer]
Make sure your installed LESS Version  is >= 3.7, the error message tells you, the consumed LESS code is unparsable,

**Reply by birthdaycorp** [reply]
thank you for giving this some attention! i'm using less 4.1.1.

i saw [this](https://marekurbanowicz.medium.com/how-to-customize-fomantic-ui-with-less-and-webpack-applicable-to-semantic-ui-too-fbf98a74506c), but it does not entirely apply. my semantic was previously loading with sprockets, not webpacker

**Reply by lubber-de** [reply]
We have a fix for LESS 4.x in #1819, but this is still in the nightly beta build 2.8.8
Please try to use that version via
`npm install fomantic-ui@nightly`
and see if it solves your issue

**Reply by birthdaycorp** [reply]
When using less 4.1.1 and installing the nightly beta build with `npm install fomantic-ui@nightly`, i get this: `npm ERR! Error: less version 4.1.1 is not currently supported`

When using anything between less 3.7.0 and 4.0.0 and installing the nightly beta build, i get this output: 

```
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm ERR! path /Users/me/code/project/node_modules/fomantic-ui
npm ERR! command failed
npm ERR! signal SIGINT
npm ERR! command sh -c gulp install
npm ERR! [02:35:34] Using gulpfile ~/code/project/node_modules/fomantic-ui/gulpfile.js
npm ERR! [02:35:34] Starting 'install'...
npm ERR! 
[02:35:34] Starting 'run setup'...
npm ERR! ? Set-up Semantic UI (Use arrow keys)
npm ERR! ❯ Automatic (Use default locations and all components) 
npm ERR!   Express (Set components and output folder) 
npm ERR!   Custom (Customize all src/dist values) [02:35:34] The following tasks did not complete: install, run setup
npm ERR! [02:35:34] Did you forget to signal async completion?

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/me/.npm/_logs/2021-02-15T10_35_35_938Z-debug.log
```


What is that `npm ERR! path /Users/me/code/project/node_modules/fomantic-ui` about?

**Reply by lubber-de** [reply]
Try to use use `npx gulp build` instead of `gulp install`

Official support for LESS 4.1.x has to be tested by ourselves first and then we can raise the  the dependency. LESS 4.1.1 is just out since 2 weeks, so i'll test it

**Reply by lubber-de** [reply]
and node >= 15 is also not supported (it breaks tons of things even in other projects than fomantic...)
See #1859

**Reply by lubber-de** [reply]
You can temporary test your LESS 4.1 by adjusting the `package.json` and change `    "less": "^3.7.0",` to  `"less": "^4.0.0",` and try `npx gulp build` again

**Reply by birthdaycorp** [reply]
i finally have fomantic-ui installed! i'm using LESS 3.13.1. it's still not working, though.😫 and i am still getting that unrecognized input error.

i ended up following [these instructions](https://www.reddit.com/r/rails/comments/l06wnq/guide_fomantic_semantic_ui_from_scratch_in_rails/), for loading with webpacker. i ran `npx gulp build` from inside **node_modules/fomantic-ui**

i have two aliases set up in **config/webpack/environment.js**

```
alias: {
      './themes': path.resolve(__dirname, '../../semantic/dist/themes'),
      'semantic': path.resolve(__dirname, '../../semantic')
    }

```
and this in **app/javascript/packs/application.js**

```
import 'jquery/src/jquery'
import 'fomantic-ui'
require('stylesheets/application.scss')
import 'semantic/dist/semantic.js'
require('semantic/dist/semantic.css')
```

**Reply by lubber-de** [reply]
Let's concentrate on the fomantic build as i dont have experience with webpacker or rails. (btw when using rails, there exists a Rails/SASS Repo (maintained by other guys) which might work better for you?  https://github.com/fomantic/Fomantic-UI-SASS

So, main question:
**Does `npx gulp build` on CLI inside the fomantic-ui root folder work?** (Otherwise there wouldn't be any dist/semantic.css) ?
If this still gives that `unrecognized input` error then 

- Which node version are you using? `node -v` Fomantic does only support node up to v14
- do you use a custom theme ? (so did you change semantic.json ?)
- Are you sure you use LESS 3.13? (try `npm list less` inside the fomantic-ui root folder)

if you dont need a custom theme at all, you may also just use the precompiled semantic.js/css from the dist folder (or use a CDN)

**Reply by birthdaycorp** [reply]
`npx gulp build` inside the **node_modules/fomantic-ui** root runs the build process and creates the files in **app/assets/semantic/semantic-ui**. That location is dependent on **semantic.json** in my app root:

```
{
  "base": "app/assets/semantic/semantic-ui",
  "paths": {
    "source": {
      "config": "src/theme.config",
      "definitions": "src/definitions/",
      "site": "src/site/",
      "themes": "src/themes/"
    },
    "output": {
      "packaged": "dist/",
      "uncompressed": "dist/components/",
      "compressed": "dist/components/",
      "themes": "dist/themes/"
    },
    "clean": "dist/"
  },
  "permission": false,
  "autoInstall": true,
  "rtl": false,
  "version": "2.8.7"
}
```
Output of node -v and less inside fomantic-ui root:

`node -v`  = 13.7.0
`npm list less` = less@3.13.1 

I should also mention that the first error I dealt with after successful installation was `expected ')' got 'o'`. That was solved by removing all the "(optional)" from the `@import` statements in **app/assets/semantic/semantic-ui/src/theme.less**

**Reply by birthdaycorp** [reply]
Previously i was using the default theme, but had made some slight modifications in the .overrides files. Can I use the precompiled css/js and retain those modifications?

**Reply by lubber-de** [reply]
Depends on your overall modifications. Maybe it's possible to just add them as custom CSS right after importing the default semantic.css

**Reply by birthdaycorp** [reply]
After installing less@3.13.1 and fomantic-ui@2.8.8-beta.158, I have finally managed to get something other than `unrecognized input`!! New error is `unmatched '/*'`. Does this mean there are some css comments without a closing `*/` in the `dist/` files?

**Reply by lubber-de** [reply]
Whatever your webpack is doing, but the original dist files from fantic don't have that issue

**Reply by birthdaycorp** [reply]
i stopped using webpack for now, just want to focus on one thing at a time. 

I just switched from Semantic UI to the community maintained Fomantic UI in my Rails app, and that `unmatched '/*'` error whenever the `stylesheet_link_tags` and such start to load fomantic assets in the document head. The fomantic styles and js are pulled into my project through the following two files:

**application.css.scss**
`*= require 'semantic-ui/src/semantic`

**application.js**
`//= require semantic-ui/dist/semantic`

I notice that it works when I make both stylesheet and js load from `semantic-ui/dist`. However, doing that excludes a lot. Icons stop working. Relevant file stored in `semantic-ui/src` is **semantic.less**. Relevant files stored in `semantic-ui/dist` are **semantic.min.css** (which i do not require except for just now when switching things around to figure this out) and **semantic.min.js** (which i do require).

Again, it "works" if I require the minified stylesheet in `semantic-ui/dist`, and it does not "work" if i require the **semantic.less** file in `semantic-ui/src`. Is this a problem with **semantic.less** or one of the other generated import files? What could this `unmatched '/*'` mean?
